### PR TITLE
change: drop support for enable_http2 and listen_port in apisix.ssl

### PR DESCRIPTION
**Pull request by tzssangglass**
### Description

Fixes #7482

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Review by tokers — APPROVED** [review]

**Review by starsz — DISMISSED** [review]

File: conf/config-default.yaml [hunk 1]
```diff
@@ -133,14 +133,13 @@ apisix:
   ssl:
     enable: true
     listen:                       # APISIX listening port in https.
-      - 9443
+      - port: 9443
+        enable_http2: true
     #   - port: 9444
```
Comment (spacewander) [inline conf/config-default.yaml:138]: We can remove this comment now as it is enabled

File: apisix/cli/ops.lua [hunk 2]
```diff
@@ -428,46 +428,28 @@ Please modify "admin_key" in conf/config.yaml .
...
-                end
-            end
-
-            if port == nil then
-                port = 9443
-            end
-
-            if enable_http2 == nil then
-                enable_http2 = false
+        local ip = value.ip
```
Comment (spacewander) [inline apisix/cli/ops.lua:431]: Let's add a check in https://github.com/apache/apisix/blob/86e9a8ea1022c34a3d5550b54a37994f174847a2/apisix/cli/schema.lua#L206 too

File: apisix/cli/ngx_tpl.lua [hunk 3]
```diff
@@ -582,9 +582,6 @@ http {
         {% if proxy_protocol and proxy_protocol.listen_http_port then %}
         listen {* proxy_protocol.listen_http_port *} default_server proxy_protocol;
         {% end %}
-        {% if proxy_protocol and proxy_protocol.listen_https_port then %}
```
Comment (spacewander) [inline apisix/cli/ngx_tpl.lua:585]: We can't simply delete the listen_https_port feature. Maybe we should provide an implementation without http2?

Comment (tzssangglass) [reply apisix/cli/ngx_tpl.lua:585]: my mistake, just remove ssl.enable_http2

File: apisix/cli/ops.lua (hunk 2 above)
Comment (tzssangglass) [reply apisix/cli/ops.lua:431]: added

File: conf/config-default.yaml (hunk 1 above)
Comment (tzssangglass) [reply conf/config-default.yaml]: remove it

File: apisix/cli/schema.lua [hunk 4]
```diff
@@ -208,6 +208,25 @@ local config_schema = {
                     properties = {
                         ssl_trusted_certificate = {
                             type = "string",
+                        },
+                        listen = {
+                            type = "array",
+                            items = {
+                                type = "object",
+                                properties = {
```
Comment (spacewander) [inline apisix/cli/schema.lua:216]: Do we need to use `required` for some fields?

Comment (tzssangglass) [reply apisix/cli/schema.lua:216]: done

Comment (tzssangglass) [reply apisix/cli/schema.lua:216]: many test cases doesn't have sll config, so ignore `required `

Comment (spacewander) [reply apisix/cli/schema.lua:216]: Yes, you are right. The ip and port are not required in the code. I made a mistake.

**Review by spacewander — APPROVED** [review]

**Review by starsz — APPROVED** [review]

**Pull request merged**